Libraries/LibDNS/Message.cpp

```diff
@@ -1005,11 +1096,36 @@ ErrorOr<Records::DNSKEY> Records::DNSKEY::from_raw(ParseContext& ctx)
     // | ALGORITHM| an 8-bit value that identifies the public key's cryptographic algorithm.
     // | PUBLICKEY| the public key material.
 
+    u32 key_tag = 0;
     auto flags = static_cast<u16>(TRY(ctx.stream.read_value<NetworkOrdered<u16>>()));
+    key_tag += (bit_cast<u16>(NetworkOrdered<u16>(flags)) & 0xff) << 8;
+    key_tag += (bit_cast<u16>(NetworkOrdered<u16>(flags)) >> 8) & 0xff;
     auto protocol = TRY(ctx.stream.read_value<u8>());
+    key_tag += static_cast<u16>(protocol) << 8;
     auto algorithm = static_cast<DNSSEC::Algorithm>(static_cast<u8>(TRY(ctx.stream.read_value<u8>())));
+    key_tag += static_cast<u16>(algorithm);
```

Comment:
you could replace the `auto`s with explicit types here.
(or even hope that `<<` coerces to int, which I am not sure about)

Reply:
They're used later with the deduced type, can't really change the variable types like that without adding an extra static_cast later.